mgail.py

```diff
@@ -109,8 +150,16 @@ def policy_loop(state_, t, total_cost, total_trans_err, _):
                 action = common.gumbel_softmax_sample(logits=mu, temperature=self.temp)
 
             # minimize the gap between agent logit (d[:,0]) and expert logit (d[:,1])
-            d = self.discriminator.forward(state_, action, reuse=True)
-            cost = self.al_loss(d)
+
+            # MODIFIED DISCRIMINATOR SECTION:
+            if self.use_irl:
+                self.discrim_output, log_p_tau, log_q_tau, log_pq = self.discriminator.forward(state_, action, self.lprobs, reuse=True)
```

Comment:
we need to pass lprobs to the feed_dict when training the policy

Reply:
Wait why does the policy need lprobs? Where would it use it?

Reply:
Yea i think the policy would output action_probs but it doesn't need lprobs as input?

Reply:
But lprobs is actually the log_q_tau, which will be needed to calculate the reward. The problem is just whether it should  involve both expert's and agent's or just agent's prob.

Reply:
Just in case you don't realize, this line is inside the policy loop, so the lprobs is used in training policy

Reply:
or maybe the lprobs in this line should be the log(action)?

Reply:
In AIRL, they use expert and agent lprobs:

`lprobs_batch = np.expand_dims(np.concatenate([lprobs_batch, expert_lprobs_batch], axis=0), axis=1).astype(np.float32)`